Fix clamp issues

forivall opened the pull request:
Fixes #188

79aafd07fc49e09708d7475d7b3ccb19ede061b7 is a rewrite of christian-burger/redshift@c611b95ea0c47b459900929ba7828db2e006cec7

----
**Commit Messages:**
83bf2a37ad617372b9c5dc8ab83cf535afb66321

    Fix CLAMP macro by renaming it to BETWEEN

    Fixes #188

    When I moved the macro definition below all of the includes, I got the
    following compiler warning:

    /usr/include/glib-2.0/glib/gmacros.h:246:0: note: this is the location
    of the previous definition
     #define CLAMP(x, low, high)  (((x) > (high)) ? (high) : (((x) < (low))
    ? (low) : (x)))

    So, that's the macro definition that was being used whenever "CLAMP" was
    used. It's pretty obvious why that's wrong.

79aafd07fc49e09708d7475d7b3ccb19ede061b7

    Fix temperature getting out of bounds

    bug was introduced by refactoring at
    jonls/redshift@bb61a25e880827e5e0f75582b5303bcbb511d3c2

    elevation was allowed to sink or raise out of bounds of transition->low
    and transition->high which lead to rather funny colors when gamma
    exceeded its bounds too

forivall commented:
I suppose I could have left it named `CLAMP` and just undef'd it first. Oh well, I don't write a ton of C, so i don't really remember everything :smile:.

jonls commented:
I merged the commit from cristian-burger in 731a7d2cd5e667a4128800948d02dd60d775be13. I guess the interaction with the glib macros means that it does not work in all cases. Since glib is only required for some of the optional components we cannot simply rely on glib to provide these macros. I would prefer not to change the name to `BETWEEN` since `CLAMP` better describes what the macro does. Undef'ing the macros is probably the best solution.

forivall commented:
I renamed it back to `CLAMP` and removed the re-application of cristian-berger's fix. Do you want me to squash this and revert it to the older sytle of `CLAMP` where it just uses the `MIN` and `MAX` macros, or is this fine?

jonls commented:
Thanks, that looks good :+1:. Yeah, please squash. Since we don't need the `MIN` and `MAX` for anything else I think it is fine to remove them.

jonls commented:
Thanks!
